### PR TITLE
Add position for resource and service

### PR DESCRIPTION
**Pull request by nadeeshaan**

### CODE REVIEW

File: modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java [hunk 1]
```diff
@@ -1372,10 +1373,11 @@ public void addServiceBody(Set<Whitespace> ws) {
                 .forEach(varDef -> serviceNode.addVariable((VariableDefinitionNode) varDef));
     }
 
-    public void endServiceDef(Set<Whitespace> ws, String protocolPkg, String serviceName) {
+    public void endServiceDef(DiagnosticPos pos, Set<Whitespace> ws, String protocolPkg, String serviceName) {
         ServiceNode serviceNode = serviceNodeStack.pop();
         serviceNode.setName(createIdentifier(serviceName));
         serviceNode.setProtocolPackageIdentifier(createIdentifier(protocolPkg));
+        ((BLangService) serviceNode).pos = pos;
```
Comment (lafernando) [inline modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java]: Change the "serviceNode" variable to BLangService itself when it's created at 1377, by doing the explicit cast there.

File: modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java [hunk 2]
```diff
@@ -1385,10 +1387,11 @@ public void startResourceDef() {
         invokableNodeStack.push(resourceNode);
     }
 
-    public void endResourceDef(Set<Whitespace> ws, String resourceName, int annotCount) {
+    public void endResourceDef(DiagnosticPos pos, Set<Whitespace> ws, String resourceName, int annotCount) {
         ResourceNode resourceNode = (ResourceNode) invokableNodeStack.pop();
         resourceNode.addWS(ws);
         resourceNode.setName(createIdentifier(resourceName));
+        ((BLangResource) resourceNode).pos = pos;
```
Comment (lafernando) [inline modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java]: Same as above.

File: modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java (hunk 1 above)
Comment (nadeeshaan) [reply modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java]: Fixed

File: modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java (hunk 2 above)
Comment (nadeeshaan) [reply modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java]: Fixed

**Pull request merged**